sdk/src/helpers/createDeterministicPrivateKey.ts

```diff
@@ -0,0 +1,59 @@
+import { type WalletClient, type Transport, type Chain, type Account, type ParseAccount, keccak256, Hex } from 'viem';
+
+/**
+ * A prefix used to ensure the generated private key is unique to this specific derivation scheme.
+ * This prevents collisions with other signature-based derivation methods.
+ */
+const DERIVATION_PATH_PREFIX = 'nitrolite_state_wallet_v1';
+
+/**
+ * Generates a deterministic private key for a session based on the wallet client,
+ * adjudicator address, app address, and a unique nonce. This key can be used
+ * for actions within a specific application context (e.g., a state channel) without
+ * requiring the user to sign every action with their main wallet.
+ *
+ * @warning This function generates a key that is valid across all chains where the
+ * adjudicator and app contracts are deployed at the same addresses. This provides a
+ * consistent session address for multi-chain dApps but removes chain-specific replay
+ * protection from the key generation itself.
```

Comment:
Are we able to use different session private keys per chain in future?
I think, it is a good security practice.

Reply:
Probably you're right, but it's not aligned with the idea of having one address